assets/js/app/DataLayer.js

```diff
@@ -98,30 +88,89 @@ LocusZoom.DataLayer.prototype.reMap = function(){
         this.data = new_data.body;
-        this.postget();
     }.bind(this));
     return promise;
 };
 
+/****************
+  Color Functions
+  Singleton for accessing/storing functions to apply different color schemes to data sets
+*/
+
+LocusZoom.DataLayer.ColorFunctions = (function() {
+    var obj = {};
+    var functions = {
+        "numeric_cut": function(parameters, value){
+            var breaks = parameters.breaks;
+            var colors = parameters.colors;
+            if (value == null || isNaN(+value)){
+                return (parameters.null_color ? parameters.null_color : colors[0]);
+            }
+            var threshold = breaks.reduce(function(prev, curr){
+                if (+value < prev || (+value >= prev && +value < curr)){
+                    return prev;
+                } else {
+                    return curr;
+                }
+            });
+            return colors[breaks.indexOf(threshold)];
+        },
+        "categorical_cut": function(parameters, value){
+            if (parameters.categories.indexOf(value) != -1){
+                return parameters.colors[parameters.categories.indexOf(value)];
+            } else {
+                return (parameters.null_color ? parameters.null_color : parameters.colors[0]); 
+            }
+        }
+    };
+
+    obj.get = function(name, parameters, value) {
+        if (!name) {
+            return null;
+        } else if (functions[name]) {
+            if (typeof parameters == "undefined" && typeof value == "undefined"){
+                return functions[name];
+            } else {
+                return functions[name](parameters, value);
+            }
+        } else {
+            throw("color function [" + name + "] not found");
+        }
+    };
+
+    obj.set = function(name, fn) {
+        if (fn) {
+            functions[name] = fn;
+        } else {
+            delete functions[name];
+        }
+    };
+
+    obj.add = function(name, fn) {
+        if (functions.name) {
+            throw("color function already exists with name: " + name);
+        } else {
+            obj.set(name, fn);
+        }
+    };
+
+    obj.list = function() {
+        return Object.keys(functions);
+    };
+
+    return obj;
+})();
+
 
 /*********************
   Positions Data Layer
 */
 
-LocusZoom.PositionsDataLayer = function(){
-
-    LocusZoom.DataLayer.apply(this, arguments);  
-    this.id = "positions";
-    this.x_field = "position";
-    this.y_field = "pvalue|neglog10";
-    this.fields = ["id", this.x_field, this.y_field, "refAllele", "ld:state"];
+LocusZoom.PositionsDataLayer = function(id, layout){
 
-    this.postget = function(){
-        this.data.map(function(d, i){
-            this.data[i].ld = +d["ld:state"];
-        }.bind(this));
-        return this;
-    };
+    LocusZoom.DataLayer.apply(this, arguments);
+    this.layout = layout;
+    this.fields = ["id", "position", "pvalue|neglog10", "refAllele", "ld:state"];
```

Comment:
Agreed - because this branch was already pretty extensive in scope I didn't begin diving into abstracting data layers here. However on the [`abstract_data_layers`](https://github.com/statgen/locuszoom/compare/new_layout_syntax...abstract_data_layers?expand=1#diff-4d12c420a365f5ace0fca5d054d05d6fR276) branch (which branches off of `new_layout_syntax`) this is already implemented.